### DISCUSSION TITLE
filtering PRs that make changes to specific folders or files

### DISCUSSION THREAD

**Discussion by anilhelvaci** [Agoric/agoric-sdk#10789] [Show and tell]
## Context
I was looking for the PRs that made changes to `a3p-integrations` folder. I think it could be interesting to others. Wanted to share here.

## Prerequisites
We need a set of CLI tools to achieve this:
* `gh` (GitHub CLI)
* `jq`
* `awc`
* `wc`

### Download PR data
I narrowed down my research to look for PRs that were created after `created:>2024-04-18` which is the date for the PR (#9257) that introduced `z:acceptance` concept.

```
gh pr list -R Agoric/agoric-sdk -S "created:>2024-04-18 state:closed state:open" -L1000 --json url,files,title,number > prs_since_9257.txt
```

### Filter data
```sh
keyword="a3p-integration"
jq -r --arg key "${keyword}" '.[] | . as $parent | .files[] | select(.path | contains($key))| .path | $parent | [.number,.title] | join (" ")' < prs_since_9257.txt | awk '!seen[$0]++' > filter.txt
```

### Count the number of PRs
```sh
wc -l filter.txt
```

**Comment by LuqiPan** [comment]
🗒️  Added `-r` in the `jq` command in `Filter data` section so the lines in `filter.txt` are without quotes
